assets/js/DataGrid.js

```diff
@@ -111,12 +112,14 @@ export class EditableColumnName extends React.Component {
     this.enterEditMode = this.enterEditMode.bind(this);
     this.handleInputChange = this.handleInputChange.bind(this);
     this.handleInputBlur = this.handleInputBlur.bind(this);
-    this.handleInputKeyPress = this.handleInputKeyPress.bind(this);
+    this.handleInputKeyDown = this.handleInputKeyDown.bind(this);
     this.handleInputFocus = this.handleInputFocus.bind(this);
   }
 
-  enterEditMode() {
-    this.setState({editMode: true});
+  enterEditMode(mode=true) {
```

Comment:
Do not add a `mode` parameter here. `enterEditMode(false)` does exactly the opposite of what I expect.

Two alternatives: a new `exitEditMode()` method, or changing it to `setIsEditing(isEditing)` with no default. Either way, you're stuck with two methods. No single method name can intuitively represent the two verbs, "start editing in response to a click" and "stop editing because I am the programmer and I say so."